app/grandchallenge/algorithms/forms.py

```diff
@@ -1401,17 +1401,37 @@ def clean_set_as_default(self):
 
         return set_as_default
 
+    def clean_inputs(self):
+        inputs = self.cleaned_data.get("inputs", [])
+
+        if (
+            AlgorithmAlgorithmInterface.objects.filter(
+                algorithm=self._algorithm
+            )
+            .annotate(
+                input_count=Count("interface__inputs", distinct=True),
+                relevant_input_count=Count(
+                    "interface__inputs",
+                    filter=Q(interface__inputs__in=inputs),
+                    distinct=True,
+                ),
+            )
+            .filter(input_count=len(inputs), relevant_input_count=len(inputs))
+            .exists()
+        ):
+            raise ValidationError(
+                "An AlgorithmInterface for this algorithm with the "
+                "same inputs already exists. "
+                "Algorithm interfaces need to have unique sets of inputs."
+            )
+        return inputs
+
     def clean(self):
         cleaned_data = super().clean()
 
         inputs = cleaned_data.get("inputs", [])
         outputs = cleaned_data.get("outputs", [])
 
-        if not inputs or not outputs:
-            raise ValidationError(
-                "You must provide at least 1 input and 1 output."
-            )
-
```

Comment:
As far as I can tell this check is unnecessary. The fields in the form are required and make it impossible to submit without at least one selected choice. Also, the create() method later on will raise an error if there are no inputs or outputs.

Reply:
The create() method actually doesn't raise an error, so I left this check in the form after all. I have also added a check to the create method to cover us from all sides.

Reply:
The check for duplicates in the input-output combination is actually redundant now that we enforce uniqueness on inputs, so I have removed that check as well.